base/simdloop.jl

```diff
@@ -125,12 +126,12 @@ either case, your inner loop should have the following properties to allow vecto
 * No iteration ever waits on a previous iteration to make forward progress.
 """
 macro simd(forloop)
-    esc(compile(forloop, nothing))
+    esc(compile(forloop, Symbol("julia.ivdep.end")))
```

Comment:
Can we keep this `nothing`? Makes little sense to have non-matching "begin"/"end" construct

Reply:
Didn't know that we have `Base.Experimental.Const` and `@aliasscope`, which seems a better tool to make `a .+= 1` vectorlizable.
So the very first motivation for this PR no longer holds: (If user wants partial `ivdep`, they could use `@aliasscope` instead in most cases.)
Then the second question is, can we use `Experimental` api in our `Base`?

Reply:
IIUC, we'd better have something like `Expr(:ivdepscope)` and `Expr(:popivdepscope)` instead.
(and add a `@ivdep` macro ?).